pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -124,34 +203,92 @@ func configMapDeleted(obj interface{}, c *K8sOrchestrator) {
 		log.Warnf("configMapDeleted: unrecognized object %+v", obj)
 		return
 	}
-	if fssConfigMap.Name == featureStatesConfigMapName && fssConfigMap.Namespace == featureStatesConfigMapNamespace {
-		for featureName := range c.featureStates {
-			c.featureStates[featureName] = strconv.FormatBool(false)
+	// Supervisor FSS configmap
+	if fssConfigMap.Name == supervisorFSSConfigMapName && fssConfigMap.Namespace == supervisorFSSConfigMapNamespace {
+		for featureName := range k8sOrchestratorInstance.supervisorFeatureStates {
+			k8sOrchestratorInstance.supervisorFeatureStates[featureName] = strconv.FormatBool(false)
 		}
-		log.Infof("configMapDeleted: %v deleted. Setting feature state values to false %v", fssConfigMap.Name, c.featureStates)
+		log.Infof("configMapDeleted: %v deleted. Setting supervisor feature state values to false %v", fssConfigMap.Name,
+			k8sOrchestratorInstance.supervisorFeatureStates)
+	}
+	// Internal FSS configmap
+	if fssConfigMap.Name == internalFSSConfigMapName && fssConfigMap.Namespace == internalFSSConfigMapNamespace {
+		for featureName := range k8sOrchestratorInstance.internalFeatureStates {
+			k8sOrchestratorInstance.internalFeatureStates[featureName] = strconv.FormatBool(false)
+		}
+		log.Infof("configMapDeleted: %v deleted. Setting internal feature state values to false %v", fssConfigMap.Name,
+			k8sOrchestratorInstance.internalFeatureStates)
 	}
 }
 
-// updateFSSValues updates feature state switch values in the k8sorchestrator
-func updateFSSValues(ctx context.Context, fssConfigMap *v1.ConfigMap, c *K8sOrchestrator) {
-	log := logger.GetLogger(ctx)
-	c.featureStates = fssConfigMap.Data
-	log.Infof("New feature states values stored successfully: %v", c.featureStates)
-}
-
-// IsFSSEnabled checks if feature state switch is enabled for the given feature indicated by featureName
+// IsFSSEnabled utilises the cluster flavor to check their corresponding FSS maps and returns
+// if the feature state switch is enabled for the given feature indicated by featureName
 func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string) bool {
```

Comment:
It should be easy to add an unit test. Can we add a unit test for this?

Reply:
Done. Added unittests for IsFSSEnabled